api/src/main/java/org/openmrs/api/db/hibernate/HibernateConditionDAO.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db.hibernate;
+
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.Query;
+import org.hibernate.SessionFactory;
+import org.openmrs.Condition;
+import org.openmrs.Patient;
+import org.openmrs.api.db.ConditionDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Hibernate implementation of the ConditionDAO
+ *
+ * @see ConditionDAO
+ * @see org.openmrs.api.ConditionService
+ */
+public class HibernateConditionDAO implements ConditionDAO {
+	
+	private static final Log log = LogFactory.getLog(ConditionDAO.class);
+	
+	/**
+	 * Hibernate session factory
+	 */
+	private SessionFactory sessionFactory;
+	
+	/**
+	 * Set session factory
+	 *
+	 * @param sessionFactory
+	 */
+	public void setSessionFactory(SessionFactory sessionFactory) {
+		this.sessionFactory = sessionFactory;
+	}
+	
+	/**
+	 * Saves the condition.
+	 *
+	 * @param condition the condition to save.
+	 * @return the saved condition.
+	 */
+	@Override
+	@Transactional
+	public Condition saveCondition(Condition condition) {
+		sessionFactory.getCurrentSession().saveOrUpdate(condition);
+		return condition;
+	}
+	
```

Comment:
Don't you need a method that looks up a condition by id?

Reply:
Done. Thanks